File: src/main/java/com/becoder/service/ProductService.java
package com.becoder.service;

import java.util.List;

import com.becoder.model.Product;

public interface ProductService {

	public Product saveProduct(Product product);
	
	public List<Product> getAllProduct();
	
	public Product getProductById(Integer id);
	
	public String deleteProduct(Integer id);
	
	
}

File: frontend/src/Service/productservice.js
import axios from "axios";

const API_URL = "http://localhost:8082"; 

class ProductService {
  

  saveProduct(product) {
    return axios.post(API_URL + "/saveProduct", product);
  }

  
  getAllProducts() {
    return axios.get(API_URL+"/");
  }

  getProductById(id) {
    return axios.get(API_URL+"/"+id);
  }

  deleteProduct(id) {
    return axios.delete(API_URL + "/deleteProduct/" + id); //  Fixed
  }

//   editProduct(product) {
//     return axios.put(API_URL + "/editProduct/", product); //Changed to PUT request
//   }
editProduct(product) {
    return axios.put(API_URL + `/editProduct/${product.id}`, product); // Pass ID in the URL
  }
  
}

export default new ProductService();
